s3fs/core.py

```diff
@@ -142,7 +169,10 @@ def _ls(self, path, refresh=False):
                     f['Size'] = 0
                     del f['Name']
             else:
-                files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                try:
+                    files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                except ClientError:
+                    files = []
```

Comment:
My experience with `distributed.s3` is that I frequently make mistakes writing down the bucket/key of my data and only realize the mistake far later on, when I find that my bag/dataframe is empty.  I these cases I would have preferred receiving a `FileNotFound` exception early.  I think that erring when performing read operations on non-existent locations is also the standard expectation within other projects as well.